tests/posix/spinlock/testcase.yaml

```
@@ -0,0 +1,28 @@
+common:
+  filter: not CONFIG_NATIVE_LIBC
+  tags:
+    - posix
+    - spinlock
+  # 1 tier0 platform per supported architecture
+  platform_key:
+    - arch
+    - simulation
+tests:
+  portability.spinlock.rwlocks:
```

Please match the option group name for all testsuites below
```suggestion
  portability.posix.spinlocks:
```